src/reactjs/useBaseQuery.ts

```diff
@@ -59,12 +59,6 @@ export function useBaseQuery<
     if (typeof defaultedOptions.staleTime !== 'number') {
       defaultedOptions.staleTime = 1000
     }
-
-    // Set cache time to 1 if the option has been set to 0
-    // when using suspense to prevent infinite loop of fetches
-    if (defaultedOptions.cacheTime === 0) {
-      defaultedOptions.cacheTime = 1
-    }
```

Comment:
@arnaudbzn I think I finally found a way to remove this suspense workaround, albeit by adding another workaround.  Can you maybe review this PR? I've left a bit of information in the commit messages